### PR TITLE
Fix '`testcafe -b` returns the "The specified "true" browser provider was not found" error'

### PR DESCRIPTION
**Pull request by Farfurix**
### Todo
- [ ] Close https://github.com/DevExpress/testcafe/issues/4294 after merge
- [x] ~~`_setProviderName` naming~~
- [x] ~~Test provider name to parse: `saucelabs` --> `providerName`?~~

https://github.com/DevExpress/testcafe/issues/4294

### Changes
1. Set the default "locally-installed" provider name in the `CliArgumentParser.parse` function.

### CODE REVIEW

**Comment by Farfurix** [comment]
@testcafe-build-bot retest

File: src/cli/argument-parser.ts [hunk 1]
```diff
@@ -260,8 +260,7 @@ export default class CLIArgumentParser {
     }
 
     private _setUpProviderName (): void {
-        if (this.opts.listBrowsers)
-            this.opts.providerName = this.opts.listBrowsers as string;
+        this.opts.providerName = (this.opts.listBrowsers === true ? void 0 : this.opts.listBrowsers) as string;
```
Comment (miherlosev) [inline src/cli/argument-parser.ts]: I don't remember the use case for which this code is necessary.
If you can call to mind the use case, please write a test for that.
If not, then just remove the `_setUpProvideName` function.

Comment (AndreyBelym) [reply src/cli/argument-parser.ts]: You can run `testcafe --list-browsers` to get locally installed browsers or `testcafe --list-browsers providerName` to get browsers supported by a `providerName` provider.

In the first case, `listBrowsers` is `true`, in the second - `"providerName"`.

Let's just write a test for the expected behaviour, and then just specify a default value for the `listBrowsers` option: https://github.com/tj/commander.js/#default-option-value

**Comment by Farfurix** [comment]
@testcafe-build-bot retest

File: src/cli/argument-parser.ts [hunk 2]
```diff
@@ -259,9 +259,16 @@ export default class CLIArgumentParser {
             this.opts.videoEncodingOptions = await getVideoOptions(this.opts.videoEncodingOptions as string);
     }
 
-    private _setUpProviderName (): void {
-        if (this.opts.listBrowsers)
-            this.opts.providerName = this.opts.listBrowsers as string;
+    private _setProviderName (): void {
```
[the remark this answers is not in the document]
Comment (VasilyStrelyaev) [reply src/cli/argument-parser.ts]: I'm thinking of `_parseBrowserSource` or smth

**Pull request merged**